lib/classes/Variables.js

```diff
@@ -40,7 +40,7 @@ class Variables {
 
     _.deepMapValues(this.service, (property, propertyPath) => {
       if (typeof property === 'string') {
-        const populateSingleProperty = new Promise((resolve) => this
+        const populateSingleProperty = new BbPromise((resolve) => this
```

Comment:
I also think that the original code can cause unhandled rejections as there's no `reject` callback defined. If there's a rejection in `populateProperty()` then this little call would never finish as `resolve()` is never triggered IMO. I definitely suggest to rework the code into something similar to what I suggested above.